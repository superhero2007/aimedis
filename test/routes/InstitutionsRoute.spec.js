import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import InstitutionsRoute from 'routes/InstitutionsRoute';

describe('routes/InstitutionsRoute', () => {
  it('should render the Institutions Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/institutions']}>
        <InstitutionsRoute
          exact
          path="/institutions"
          component={() => (<div>Institutions</div>)}
          isAuthenticated={false}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });

  it('should render the Institutions Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/institutions']}>
        <InstitutionsRoute
          exact
          path="/institutions"
          component={() => (<div>Institutions</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
