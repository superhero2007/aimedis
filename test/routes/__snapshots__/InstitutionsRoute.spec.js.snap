// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/InstitutionsRoute should render the Institutions Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">Institutions</div>"`;

exports[`routes/InstitutionsRoute should render the Institutions Page for unauthenticated access 1`] = `"<div data-reactroot=\\"\\">Institutions</div>"`;
